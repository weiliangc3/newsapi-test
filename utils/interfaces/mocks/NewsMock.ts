import { Article, News, Source } from '../News';

export const SourceMock: Source = {
  id: 136,
  name: 'Mock Source Name',
};

export const ArticleMock: Article = {
  source: SourceMock,
  author: 'Mock aticle Author',
  title: 'Mock article name',
  description: 'Mock desc',
  url: 'https://mock.article',
  urlToImage: 'https://mock.article/image',
  publishedAt: 'mockpublishedatstring',
  content: 'Mock Article Content Of An Amazing Piece Of Journalistic Integrity',
};

export const NewsMock: News = {
  status: 'ok',
  totalResults: 20,
  articles: [
    ArticleMock,
  ],
};
