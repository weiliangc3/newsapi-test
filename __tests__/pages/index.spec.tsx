/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import '@testing-library/jest-dom';
import { useRouter } from 'next/router';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Page from '../../pages/index';
import { NewsMock } from '../../utils/interfaces/mocks/NewsMock';

const mockRouterPush = jest.fn();
jest.mock('next/router', () => ({
  useRouter: jest.fn(() => ({
    push: mockRouterPush,
  })),
}));

describe('index page', () => {
  it('renders correctly', () => {
    const tree = render(<Page news={NewsMock} />);
    expect(tree.container).toMatchSnapshot();
  });

  it('takes a search term and can be used to search', () => {
    render(<Page news={NewsMock} />);
    userEvent.type(screen.getByTestId('search-box'), 'joji');
    userEvent.click(screen.getByText('Get yer noose here'));
    expect(mockRouterPush).toHaveBeenCalledWith('/news?rope=joji');
  });
});
