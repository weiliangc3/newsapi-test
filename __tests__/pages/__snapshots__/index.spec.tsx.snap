// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index page renders correctly 1`] = `
<div>
  <h1
    class="sc-bdvvtL iSuoRM"
  >
    The Noose
  </h1>
  <p
    class="sc-gsDKAQ eLJMeh"
  >
    Life's a bitch and then you die.
  </p>
  <input
    data-testid="search-box"
    type="text"
    value=""
  />
  <button
    type="button"
  >
    Get yer noose here
  </button>
  <h2
    class="sc-dkPtRN hMeJFp"
  >
    Top news
  </h2>
  <div>
    <div
      role="button"
      tabindex="0"
    >
      Mock article name
    </div>
  </div>
</div>
`;
